cache.go

```diff
@@ -478,6 +478,12 @@ func (c *Cache[K, V]) Items() map[K]*Item[K, V] {
 // Range stops the iteration.
 func (c *Cache[K, V]) Range(fn func(item *Item[K, V]) bool) {
 	c.items.mu.RLock()
+
+	// Check if cache is empty
+	if c.items.lru.Len() == 0 {
+		return
```

Comment:
Just noticed this: you need to call `c.items.mu.RUnlock()` before returning here. We aren't using `defer` for this because the range block below needs a special kind of locking/unlocking.

Reply:
> Just noticed this: you need to call `c.items.mu.RUnlock()` before returning here. We aren't using `defer` for this because the range block below needs a special kind of locking/unlocking.

fixed that :)